### PR TITLE
Add GenerateCreateScript extension method on DatabaseFacade

### LINKED ISSUE #2943: Equivalent to ObjectContext.CreateDatabaseScript
This is one of my favorite features of the EF Power Tools, and I think we should have similar functionality in EF Core.

My proposal is an `db.Database.GenerateCreateScript()` extension method. Here's the implementation. :wink:

``` C#
public static string GenerateCreateScript(this DatabaseFacade database)
{
    var model = database.GetService<IModel>();
    var differ = database.GetService<IMigrationsModelDiffer>();
    var generator = database.GetService<IMigrationsSqlGenerator>();
    var sql = database.GetService<ISqlGenerationHelper>();

    var operations = differ.GetDifferences(null, model);
    var commands = generator.Generate(operations, model);

    var builder = new StringBuilder();
    foreach (var command in commands)
    {
        builder
            .Append(command.CommandText)
            .AppendLine(sql.BatchTerminator);
    }

    return builder.ToString();
}
```

### PR DESCRIPTION
**Pull request by ErikEJ**
fixes #2943

- also fixes a test localization bug causing test failure on non US systems.

### CODE REVIEW

File: src/EFCore.Relational/Storage/RelationalDatabaseCreator.cs [hunk 1]
```diff
@@ -265,5 +266,28 @@ public virtual bool EnsureCreated()
...
+        /// </returns>
+        public virtual string GenerateCreateScript(string batchTerminator)
+        {
+            var commands = GetCreateTablesCommands();
+            var builder = new StringBuilder();
+            foreach (var command in commands)
+            {
+                builder
+                    .Append(command.CommandText)
+                    .AppendLine(batchTerminator);
```
Comment (bricelam) [inline src/EFCore.Relational/Storage/RelationalDatabaseCreator.cs]: Any reason not to get this directly from `ISqlGenerationHelper` here?

Comment (bricelam) [reply src/EFCore.Relational/Storage/RelationalDatabaseCreator.cs]: (Adding it to `RelationalDatabaseCreatorDependencies`)

Comment (ErikEJ) [reply src/EFCore.Relational/Storage/RelationalDatabaseCreator.cs]: I actualky did this first, but got warnings about breaking changes due to constructor having been changed

Comment (smitpatel) [reply src/EFCore.Relational/Storage/RelationalDatabaseCreator.cs]: @ErikEJ - Which ctor? ctor of `RelationalDatabaseCreatorDependencies` is fine to break.

Comment (ErikEJ) [reply src/EFCore.Relational/Storage/RelationalDatabaseCreator.cs]: @smitpatel Yes, ctor of RelationalDatabaseCreatorDependencies  - I will break it then! 😄

Comment (ErikEJ) [reply src/EFCore.Relational/Storage/RelationalDatabaseCreator.cs]: @smitpatel Broken!

**Pull request merged**